pyof/foundation/basic_types.py

```diff
@@ -421,10 +421,23 @@ def __init__(self, value=b''):  # noqa
         Raises:
             ValueError: If given value is not bytes.
         """
-        if not isinstance(value, bytes):
-            raise ValueError('BinaryData must contain bytes.')
+        value = self._pack_value(value)
         super().__init__(value)
 
+    @staticmethod
+    def _pack_value(value):
```

Comment:
I am not sure you read my reply... It looks like you didn't.
I changed it anyway...